test/parse.js

```diff
@@ -65,3 +65,11 @@ test('object properties', t => {
 	t.falsy(fn.parse().prototype);
 	t.deepEqual(fn.parse('hasOwnProperty=foo'), {hasOwnProperty: 'foo'});
 });
+
+test('query strings having indexed arrays', t => {
+	t.deepEqual(fn.parse('foo[0]=bar&foo[1]=baz'), {foo: ['bar', 'baz']});
```

Comment:
Will this be in pair with `querystring`?
Cause:
```
> querystring.parse('foo[0]=bar&foo[1]=baz')
{ 'foo[0]': 'bar', 'foo[1]': 'baz' }
> querystring.parse('foo[1]=bar&foo[0]=baz')
{ 'foo[1]': 'bar', 'foo[0]': 'baz' }
```

Is this intended? Or do you mean by adding a parsing option like the stringify's `arrayFormat`?